chips/sam4l/src/usart.rs

```diff
@@ -1201,12 +1204,17 @@ impl<'a> spi::SpiMaster<'a> for USART<'a> {
 
     /// Returns the actual rate set
     fn set_rate(&self, rate: u32) -> Result<u32, ErrorCode> {
+        let non_zero_rate = match NonZeroU32::new(rate) {
+            Some(non_zero_rate) => non_zero_rate,
+            None => return Err(ErrorCode::INVAL),
+        };
         let usart = &USARTRegManager::new(self);
-        self.set_baud_rate(usart, rate);
+        self.set_baud_rate(usart, non_zero_rate);
 
         // Calculate what rate will actually be
         let system_frequency = self.pm.get_system_frequency();
-        let cd = system_frequency / rate;
+        // DIVISION: No division by 0 can occur because of the `non_zero_rate` type
```

Comment:
Does it make sense to get rid of all the potential panics here? `cd` might result in `0` if `system_frequency < non_zero_rate` so the `system_frequency / cd` will panic. Perhaps should use `checked_div` and return appropriate ErrorCode?